libs/traits/src/lib.rs

```diff
@@ -776,6 +776,8 @@ pub mod ops {
 		/// assert_eq!(extrinsic(), Err(DispatchError::Arithmetic(ArithmeticError::Overflow)));
 		/// ```
 		fn ensure_mul(&self, v: &Self) -> Result<Self, ArithmeticError> {
+			// TODO: Improve the error checking the case where the multiplication shoul be
```

Comment:
```suggestion
			// TODO: Improve the error checking the case where the multiplication should be
```

Reply:
Alternatively, you can add an error variant `ArithmeticError::OutOfBounds` which would be an union of Overflow and Underflow, which I guess would be ok in terms of error handling?

Reply:
Thanks for the review Nuno 😃 

The problem is that `ArithmeticError` is something defined by Substrate 😢 . 
I have an idea in mind. I'm going to add a new commit to it soon.

Reply:
Updated. I think I have covered all the cases.